connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -272,7 +274,14 @@ class SparkConnectPlanner(session: SparkSession) {
 
   private def transformLocalRelation(rel: proto.LocalRelation): LogicalPlan = {
     val attributes = rel.getAttributesList.asScala.map(transformAttribute(_)).toSeq
-    new org.apache.spark.sql.catalyst.plans.logical.LocalRelation(attributes)
+
+    val rows = ArrowConverters.fromBatchIterator(
+      rel.getDataList.asScala.map(_.toByteArray).iterator,
```

Comment:
If `data` is a regular byte array, it becomes a ByteString that you can simply extract here.